ci/taos/plugins-good/pr-format-sloccount.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+##
+# Copyright (c) 2018 Samsung Electronics Co., Ltd. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##
+# @file     pr-format-sloccount.sh
+# @brief    Count physical source lines of code (SLOC)
+#
+# It is a set of tools for counting physical Source Lines of Code (SLOC) in a large
+# number of languages of a potentially large set of programs.
+#
+# @see      https://packages.ubuntu.com/search?keywords=sloccount
+# @see      https://github.com/nnsuite/TAOS-CI
+# @author   Geunsik Lim <geunsik.lim@samsung.com>
+#
+
+# @brief [MODULE] TAOS/pr-format-sloccount
+function pr-format-sloccount(){
+echo "########################################################################################"
+echo "[MODULE] TAOS/pr-format-sloccount: Count physical source lines of code (SLOC)"
+pwd
+
+# Check if server administrator install required commands
+check_dependency sloccount
+
+# Read file names that a contributor modified(e.g., added, moved, deleted, and updated) from a last commit.
+FILELIST=`git show --pretty="format:" --name-only --diff-filter=AMRC`
+
+# Inspect all files that contributor modifed.
+for i in ${FILELIST}; do
+    # default value of check_result is "skip".
+    check_result="skip"
+
+    # skip obsolete folder
+    if [[ $i =~ ^obsolete/.* ]]; then
+        continue
+    fi
+    # skip external folder
+    if [[ $i =~ ^external/.* ]]; then
+        continue
+    fi
+    # Handle only text files in case that there are lots of files in one commit.
+    echo "[DEBUG] file name is ( $i )."
+    if [[ `file $i | grep "ASCII text" | wc -l` -gt 0 ]]; then
+        # in case that PR includes source codes: *.c,  *.cpp, *.py)
+        case $i in
+            *.c | *.cpp | *.py)
+                echo "[DEBUG] ( $i ) file is a document file with a ASCII text format."
+                sloc_analysis_sw="sloccount"
+                sloc_analysis_rules="--wide --multiproject"
+                sloc_check_result="sloccount_result.txt"
+
+                # run a module
+                $sloc_analysis_sw $sloc_analysis_rules $i > ../report/${sloc_check_result}
+
+                line_count=`cat ../report/${sloc_check_result} | wc -l`
```

Comment:
what is the condition of 'success' and 'failure'?

Reply:
It will be keep a "success" status in case of SLOCCount module.